File: src/exercises/ex1sphereArea/SphereArea.py
# package exercises

from math import pi


# Program to calculate the surface area of a sphere
# See https://en.wikipedia.org/wiki/Sphere#Surface_area
#
# Formula is: area = 4 * pi * radius² (m²)
# @staticmethod
def input_radius() -> float:
    return float(input("Give me a radius please > "))


# @staticmethod
def print_area_result(the_area: float):
    print(f"This is the area: {the_area}")


# @staticmethod
def calculate_area_of_sphere(the_radius):
    the_area = 4 * pi * the_radius ** 2
    return the_area


# @staticmethod
def calculate_sphere_area_program():
    the_radius = input_radius()
    the_area = calculate_area_of_sphere(the_radius)
    print_area_result(the_area)


# Recommended way to make module executable
if __name__ == "__main__":
    calculate_sphere_area_program()

File: src/exercises/ex1sphereArea/SphereArea.java
package exercises.ex1sphereArea;

import java.util.Scanner;

import static java.lang.Math.PI;
import static java.lang.Math.pow;
import static java.lang.System.in;
import static java.lang.System.out;

public class SphereArea {
    // TODO Fix the implementation by inserting the correct types!
    // Hint: IntelliJ can help you out.
    public static final Scanner input = new Scanner(in);

    public static float inputRadius() {
        out.println("Give me a radius please > ");
        theRadius = input.nextFloat();
        return theRadius;
    }

    public static void printAreaResult(theArea) {
        out.println("This is the area: " + theArea);
    }

    private static void calculateSphereAreaProgram() {
        // TODO implement program here
        theRadius = inputRadius();
        theArea = calculateArea(theRadius);
        printAreaResult(theArea);

    }

    private static float calculateArea(float radius) {
        return (float)(4 * PI * pow(radius, 2));
    }


    public static void main(String[] args) {
        calculateSphereAreaProgram();
    }


}
